.github/workflows/ci-pipeline-branch.yml

```
@@ -0,0 +1,18 @@
+name: CI Pipeline Push and Pull Request
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build-lib:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - run: npm install
+      - run: npm run build
```

I think it would be good to add `pnpm` usage and reuse this workflow. 🙂

```suggestion
on:
  push:
    branches: [master]
  pull_request:
    branches: [master]

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node: [18]
    steps:
      - uses: actions/checkout@v3
      - uses: pnpm/action-setup@v2
        with:
          version: latest
      - name: Use Node.js ${{ matrix.node }}
        uses: actions/setup-node@v3
        with:
          cache: pnpm
          node-version: ${{ matrix.node }}
      - name: Install dependencies
        run: pnpm install
      - name: Build
        run: pnpm build
```